timeline/itemgraph.go

```diff
@@ -994,6 +994,11 @@ var classifications = []Classification{
 		Labels:      []string{"Collection", "Album", "Playlist"},
 		Description: "A group of items",
 	},
+	{
+		Name:        "file",
+		Labels:      []string{"File"},
+		Description: "Filesystem file",
+	},
```

Comment:
Hmm, can we be more specific than this?

I'm curious _what your files are_, semantically speaking. Are they journal entries? Logs? Messages? Random notes? That's kind of what I'm hoping to reveal here by classifying an item. For example, if they're messages, then applying the `message` class will allow the items to show up in the Conversation view.

I suppose this could be _marginally_ more useful than no classification at all, but from what I've seen so far, the vast majority of items do come from files in one form or another. So I almost wonder if it's better to just leave the unknowns unclassified.

Reply:
> I'm curious what your files are, semantically speaking. Are they journal entries? Logs? Messages? Random notes? That's kind of what I'm hoping to reveal here by classifying an item. For example, if they're messages, then applying the message class will allow the items to show up in the Conversation view.

Makes sense, I should have elaborated the opening post a bit more, in retrospective. I was experimenting with the generic data source and have it import my .txt and .md notes.

That works for the most part, the only problem I've found is that the frontend crashes:

```
Uncaught (in promise) TypeError: groups[(groups.length - 1)] is undefined
    timelineGroups http://127.0.0.1:12002/resources/js/common-lib.js:1583
    loadAllGroups http://127.0.0.1:12002/resources/js/timeline.js:25
```

It seems that happens because the items the generic data source creates do not have a classification attached, that the frontend needs to group items when displaying the timeline (I think). Being a generic data source that imports files, I figured a `File` classification would be generic enough for that data source. Or perhaps a `ClassUnclassified`?. But leaving it unclassified makes sense to me also, it's just the fronted will need a different fix, for those items without a classification attached.

> I suppose this could be marginally more useful than no classification at all, but from what I've seen so far, the vast majority of items do come from files in one form or another. So I almost wonder if it's better to just leave the unknowns unclassified.

Indeed. The rational is that I thought `File` would be generic enough to be equivalent to unclassified for that particular data source, but on a second thought, not everything unclassified should be a file, so it's definitely not a good fit for everything outside that data source perhaps.

All that to say that, I'm not particularly attached to the solution I proposed :smile:

Reply:
I forgot. The crash can be reproduced with the python script I linked. Run it and go visit `/timeline` to find the issue.

Reply:
Ah, whoops. The UI should support items without a classification, so that's something I want to fix.

Thanks for the details! If these are notes, maybe we should make a new class to accommodate notes, 'ClassNotes' perhaps.

And on that note (heh), we could make a new Markdown data source that can perhaps try to read metadata from front matter. I doubt that is relevant for any of your notes so we don't have to do that for now, but it is on my list for later.

Anyway, if there's literally nothing special to extract from these files, maybe a Notes class is good enough for now (and fixing the UI bug).

What do you think?

Reply:
Sounds good to me!
Would you like the generic data source to add that Notes class when text/plain is detected and leave the rest uncategorized? Maybe class Text instead of Notes?

Reply:
> And on that note (heh), we could make a new Markdown data source that can perhaps try to read metadata from front matter. I doubt that is relevant for any of your notes so we don't have to do that for now, but it is on my list for later.

I'd be happy to sync my teeth there if no one is taking on the md data source.

Reply:
I think I've fixed the UI bug by always assigning an unclassified item a category. That should cause a group to be created for it (since `"unknown" != undefined`).

Reply:
Oh your new comments just showed up for me.

> Sounds good to me! Would you like the generic data source to add that Notes class when text/plain is detected and leave the rest uncategorized? Maybe class Text instead of Notes?

I don't know if we should assign a class if we aren't sure what it is. Could be a user-defined thing though, like in the data source options, "Classify text files as... [document/note/nothing]"